### ISSUE TITLE
Compute centerlines in the Modeling tool crashes 

### ISSUE THREAD

**Issue by ktbolt** [SimVascular/SimVascular#774]
I've been trying to compute centerlines from the Modeling tool for the CoronaryProject and the HealthyPulmonary models using the current master branch. For the CoronaryProject the following is printed to a terminal
```
NumSources 1
NumTargets 10
Calculating CenterlineSections...
  Generating surface normals
  Generating clean centerline
  Calculating centerline tangents
  Refining centerline at caps
  Rough coloring surface branches
  Slicing surface at 5763 centerline points
    0%
    10%
    20%
    30%
    40%
    50%
    59%
    69%
    79%
    89%
    99%
    100%
  Cleaning centerline bifurcations
  Splitting centerline in branches and bifurcations
  Coloring surface branches
  Coloring surface bifurcations
  Done!
/Users/parkerda/software/ktbolt/SimVascular/build/SimVascular-build/sv: line 153:  3800 Abort trap: 6 
```
Done indeed!

The 2020-06-04 SV release prints to the terminal
```
NumSources 1
NumTargets 10
Getting Center Lines...
Grouping Sections...
Merging Sections...
```
and does not crash.

@mrp089 You made some changes to `Code/ThirdParty/vmtk/simvascular_vmtk`. Maybe the call to the centerline extraction code in `sv4guiModelUtils::CreateCenterlines()` is releasing memory incorrectly?

**Comment by ktbolt** [comment]
The problem is that the centerlines returned from `vtkvmtkPolyDataCenterlines()` don't have **CenterlineIds** and **GroupIds** data arrays. `vtkvmtkMergeCenterlines()` is then called but does not find these arrays and just returns an empty PolyData object.  This upsets SV so it crashes when trying to convert centerlines to paths.

There is not much error checking in SV for things like this. `vtkvmtkMergeCenterlines()` does detect this error 
```
  vtkDataArray* groupIdsArray = input->GetCellData()->GetArray(this->GroupIdsArrayName);

  if (!groupIdsArray)
    {
    vtkErrorMacro(<<"GroupIdsArray with name specified does not exist");
    return 1;
    }
```
but keeps quite about it, `vtkErrorMacro()` does not seem to do anything.

**Comment by mrp089** [comment]
@ktbolt I did the following to reproduce the error:
- pulled & compiled the current master (be89d17)
- downloaded the [Coronary Project](https://simtk.org/frs/download_confirm.php/file/5115/CoronaryProject.zip?group_id=930)
- created a 1D model
- select `cap_aorta` as inlet face
- run the centerline extraction

I get the centerline as expected without errors. It reports `5656` points as opposed to your `5763` points. Also, `CenterlineIds` and `GroupIds` are not used by the 1D pipeline, so it shouldn't be a problem if they're missing.

This makes me suspicious: Is there another way of computing/using centerlines in SimVascular?

**Comment by ktbolt** [comment]
@mrp089 The Modeling tool also computes centerlines and I just recently added computing centerlines to the Colliding Fronts tool.

The Modeling tool calling sequence is
```
sys_geom_centerlines()

sys_geom_centerlinesections()

sys_geom_mergecenterlines()
```
The `sys_geom_mergecenterlines()` function expects the input centerline vtkPolyData to have cell data arrays named **GroupIds** and **CenterlineIds**. The vtkPolyData centerlines computed by the new centerlines code does not have these cell data arrays. This is causing SV to later crash when trying to extract paths from the merged centerlines.

We don't need to creates paths in the Modeling tool (although I will do this in the Colliding Fronts tool) but we need to put back the old `sys_geom_centerlines()` and  `sys_geom_centerlinesections()` functions to maintain the previous format in case users have been using it, rename your new functions and use them for creating 1D models. 

Comments @osmsc ?

**Comment by mrp089** [comment]
I see! I'll fix that.

**Comment by mrp089** [comment]
Fixed in d4d9473

**Comment by ktbolt** [comment]
The fixes have been merged and I tested.
